### PR TITLE
allow AWS_SESSION_TOKEN to create session token

### PR DESCRIPTION
**Pull request by joohae-kim**
for me, it's not easy to explaining the detail why I wanted merge the update into the main repo
however, I think we can use AWS_SESSION_TOKEN variable with simple touch up
please, refer following issue ticket
thank you

Reference: https://github.com/fastlane-community/fastlane-plugin-s3/issues/54

### CODE REVIEW

File: lib/fastlane/plugin/aws_s3/actions/aws_s3_action.rb [hunk 1]
```diff
@@ -89,7 +91,7 @@ def self.run(config)
         client_cfg[:region] = s3_region if s3_region
         client_cfg[:endpoint] = s3_endpoint if s3_endpoint
         client_cfg[:profile] = s3_profile if s3_profile
-        client_cfg[:credentials] = Aws::Credentials.new(s3_access_key, s3_secret_access_key) if s3_access_key && s3_secret_access_key
+        client_cfg[:credentials] = Aws::Credentials.new(s3_access_key, s3_secret_access_key, s3_session_token) if s3_access_key && s3_secret_access_key
```
Comment (joohae-kim) [inline lib/fastlane/plugin/aws_s3/actions/aws_s3_action.rb:94]: https://docs.aws.amazon.com/sdk-for-ruby/v2/api/Aws/Credentials.html

the default value of the session token filed is null, hence it should be okay

**Comment by kondratk** [comment]
@joohae-kim I've tested your PR and it seems like working fine. 
@joshdholtz is there any chance for merging this one soon?

**Comment by paul-verkada** [comment]
@joshdholtz bump

**Comment by brunomgsantos** [comment]
@joshdholtz this feature could be really helpful for AWS Federated Logins 👍

**Comment by seanedwards** [comment]
+1, we just ran into this as well. We use SAML-federated logins to AWS, so all of our development credentials require `AWS_SESSION_TOKEN`

**Review by joshdholtz — APPROVED** [review]
This look great! Thank you so much for this addition 🙌  This will go out today!

**Pull request merged**